<a href="https://colab.research.google.com/github/Supratik27/Motivational-Poster/blob/main/Naive_Bayers_classifiers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt   #pyplot helps matplotlib to work on matlab environmnet

Lets download the iris dataset from its weblink as follows:

In [ ]:
path = "https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data"

Assigning the column names to the dataset

In [ ]:
headernames = ["sepal-lenght","sepal-width","petal-lenght","petal-width","class"]

We have to read the dataset using pandas data frame

In [ ]:
data = pd.read_csv(path, names=headernames)
data.shape

(150, 5)

Now,we observe the data

In [ ]:
data.head() #used for observation of first 5-6 datasets in case of lenghty dataset

sepal-lenght  sepal-width  petal-lenght  petal-width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

Implementing steps of data preprocessing

In [ ]:
X = data.iloc[:,:-1].values
y = data.iloc[:,4].values

Diving the data into train and test splits,with 70% of the training data and 30% of the testing data

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.30)

#Data scaling

from sklearn.preprocessing import StandardScaler  #Used for data normalization(data-mean/standard deviation)
scaler = StandardScaler()
scaler.fit(X_train)
X_train = scaler.transform(X_train)
X_test = scaler.transform(X_test)

We will train the model with the help of the GaussianNB class of sklearn

In [ ]:
from sklearn.naive_bayes import GaussianNB
classifier = GaussianNB()
classifier.fit(X_train, y_train);

Now,we will make the prediction

In [ ]:
y_pred = classifier.predict(X_test)

Let's print the results as follows:

In [ ]:
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
result = confusion_matrix(y_test, y_pred)
print("The confusion matrix:")
print(result)
result1 = classification_report(y_test, y_pred)
print("The classification report:",)
print(result1)
result2 = accuracy_score(y_test, y_pred)
print("The accuracy score is:", result2)

The confusion matrix:
[[13  0  0]
 [ 0 12  2]
 [ 0  1 17]]
The classification report:
                 precision    recall  f1-score   support

    Iris-setosa       1.00      1.00      1.00        13
Iris-versicolor       0.92      0.86      0.89        14
 Iris-virginica       0.89      0.94      0.92        18

       accuracy                           0.93        45
      macro avg       0.94      0.93      0.94        45
   weighted avg       0.93      0.93      0.93        45

The accuracy score is: 0.9333333333333333


Using the Naive Bayes classifier, we have achieved an accuracy of 93.33% (approximately).